MapboxNavigationUI/MGLMapView.swift

```diff
@@ -145,31 +153,21 @@ extension MGLMapView {
             arrowStroke.lineWidth = MGLStyleValue(rawValue: 8)
             arrowStroke.lineColor = MGLStyleValue(rawValue: NavigationUI.shared.tintColor)
             
-            style?.addSource(arrowSourceStroke)
-            style?.addSource(arrowSource)
+            if let source = style.source(withIdentifier: arrowSourceIdentifier) {
+                let s = source as! MGLShapeSource
+                s.shape = MGLShapeCollection(shapes: maneuverArrowPolylines)
+            } else {
+                style.addSource(arrowSource)
+                style.addLayer(arrow)
```

Comment:
Move all the code that sets up `arrowSource` and `arrow` into this `else` block. That way it’s clear that, on the second time around, the only thing that happens is the existing source’s shape getting modified.